src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/MicrosoftQualityGuidelinesAnalyzersResources.resx

```diff
@@ -252,4 +252,10 @@
   <data name="AvoidPropertySelfAssignmentMessage" xml:space="preserve">
     <value>The property {0} should not be assigned to itself.</value>
   </data>
+  <data name="ReferingToObjectAndReassigningItInTheSameStatementMessage" xml:space="preserve">
+    <value>Object '{0}' is being referred to and reassigned in the same statement. You are at risk of referring to unintended object.</value>
```

Comment:
How about:

```suggestion
    <value>Object '{0}' is referenced and reassigned in the same statement, which means the reference is indeterminate.</value>
```

Reply:
Changed this string to your version. Thanks for help